cmd/parse/gov/proposal.go

```diff
@@ -124,7 +125,11 @@ func refreshProposalDetails(parseCtx *parser.Context, proposalID uint64, govModu
 
 	// Handle the MsgSubmitProposal messages
 	for index, msg := range tx.GetMsgs() {
-		if _, ok := msg.(*govtypesv1.MsgSubmitProposal); !ok {
+		_, isMsgSubmitProposalV1 := msg.(*govtypesv1.MsgSubmitProposal)
+		_, isMsgSubmitProposalV1Beta1 := msg.(*govtypesv1beta1.MsgSubmitProposal)
+
+		// Skip if the message is not a submit proposal message
+		if !isMsgSubmitProposalV1 && !isMsgSubmitProposalV1Beta1 {
 			continue
 		}
 
```

Comment:
What about

```go
		switch msg.(type) {
		case *govtypesv1.MsgSubmitProposal, *govtypesv1beta1.MsgSubmitProposal:
			err = govModule.HandleMsg(index, msg, tx)
			if err != nil {
				return fmt.Errorf("error while handling MsgSubmitProposal: %s", err)
			}

		}
```

Reply:
Great, it is cleaner, thanks!
https://github.com/forbole/callisto/pull/725/commits/3936cc6693dc527c9ba1ddc0317794902ccbdcab